hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java

```diff
@@ -403,7 +403,7 @@ public void setOrigin(FromElement origin, boolean manyToMany) {
 				ASTUtil.appendSibling( origin, this );
 			}
 			else {
-				if ( !getWalker().isInFrom() && !getWalker().isInSelect() ) {
+				if ( !getWalker().isInFrom() && !getWalker().isInSelect() && !getWalker().isInEntityGraph()) {
```

Comment:
I am not understanding this part of the change.   If we are not setting the origin here, where do we?